### ISSUE TITLE
dict-iter-method misses .iter*() of a defaultdict()

### ISSUE THREAD

**Issue by edmorley** [PyCQA/pylint#2005]
### Steps to reproduce
1. `git clone https://github.com/mozilla/treeherder.git`
2. `cd treeherder && git checkout addd73c8631ee2c005a0139bf0a88fabe3c6a24b`
3. `rm pylintrc`
4. `pylint -d all -e dict-iter-method -s n treeherder/autoclassify/management/co mmands/es_benchmark.py`

### Current behavior
```
************* Module treeherder.autoclassify.management.commands.es_benchmark
W:100,27: Calling a dict.iter*() method (dict-iter-method)
```

### Expected behavior
In addition to the instance found on line 100, `dict-iter-method` would ideally catch the `iteritems()` being performed on an instance of `defaultdict(list)` on line 84:
https://github.com/mozilla/treeherder/blob/addd73c8631ee2c005a0139bf0a88fabe3c6a24b/treeherder/autoclassify/management/commands/es_benchmark.py#L84

### pylint --version output
```
No config file found, using default configuration
pylint 1.8.4,
astroid 1.6.3
Python 2.7.14 (default, Sep 26 2017, 13:42:32)
[GCC 5.4.0 20160609]
```

**Comment by edmorley** [comment]
Another variation that would presumably be fixed by the same fix as above: custom subclasses of `dict()`.

### Steps to reproduce
1. `git clone https://github.com/mozilla/treeherder.git`
2. `cd treeherder && git checkout addd73c8631ee2c005a0139bf0a88fabe3c6a24b`
3. `rm pylintrc`
4. `pylint -d all -e dict-iter-method -s n treeherder/client/thclient/perfherder.py`

### Current behavior
No warnings reported.

### Expected behavior
For `dict-iter-method` to report on the `self.iteritems()` here:
https://github.com/mozilla/treeherder/blob/addd73c8631ee2c005a0139bf0a88fabe3c6a24b/treeherder/client/thclient/perfherder.py#L49

...since `self` is a subclass of `dict`:
https://github.com/mozilla/treeherder/blob/addd73c8631ee2c005a0139bf0a88fabe3c6a24b/treeherder/client/thclient/perfherder.py#L29

### pylint --version output
```
No config file found, using default configuration
pylint 1.8.4,
astroid 1.6.3
Python 2.7.14 (default, Sep 26 2017, 13:42:32)
[GCC 5.4.0 20160609]
```

**Comment by PCManticore** [comment]
Thanks for the report! We could definitely look at subclasses of dicts & friends.

**Comment by PCManticore** [comment]
Thanks, this is now fixed in the 1.9 branch.
